### PR TITLE
Add JUnit 5 support and remove dependency on legacy parent pom

### PR DESCRIPTION
**Pull request by uguy**
Proposed improvements on maven aspects

### CODE REVIEW

File: src/main/resources/archetype-resources/src/main/resources-filtered/__artifactId__.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Custom.category=Custom connectors
-connectorDefinitionDescription=
-connectorDefinitionLabel=${artifactId}
+connectorDefinitionDescription=Retrieve data about a star wars character
```
Comment (alachambre) [inline src/main/resources/archetype-resources/src/main/resources-filtered/__artifactId__.properties:2]: We can't put star wars stuff in the generated project. 
The default description should be something like _Indicate in a few words the purpose of this connector_

**Comment by alachambre** [comment]
It looks fine. 
I tested it with groovy, it seems to work correctly, did you try with the other languages? (generate - build - import and execute in Bonita)? 

**The generated readme AND the documentation on the Bonita website must be updated**, impl and def are not in resources anymore but in resources-filtered.

File: src/main/resources/archetype-resources/kotlin-pom.xml [hunk 2]
```diff
@@ -1,105 +1,175 @@
...
-			<version>${kotlin.version}</version>
-		</dependency>
-		<dependency>
-			<artifactId>junit</artifactId>
-			<groupId>junit</groupId>
-			<scope>test</scope>
-		</dependency>
-	</dependencies>
+        <!-- Bonita -->
+        <bonita.engine.version>7.10.0</bonita.engine.version>
```
Comment (rbioteau) [inline src/main/resources/archetype-resources/kotlin-pom.xml:28]: use property

File: src/test/resources/projects/testGroovyProject/reference/pom.xml [hunk 3]
```diff
@@ -1,120 +1,165 @@
...
+
+        <!-- Connector -->
+        <connector-definition-id>${project.artifactId}</connector-definition-id>
+        <connector-definition-version>1.0</connector-definition-version>
+        <connector-impl-id>${connector-definition-id}-impl</connector-impl-id>
+        <connector-impl-version>${project.version}</connector-impl-version>
+        <connector-main-class>com.company.bonitasoft.MyGroovyConnector</connector-main-class>
+
+        <!-- Bonita -->
         <bonita.engine.version>7.10.0</bonita.engine.version>
```
Comment (rbioteau) [inline src/test/resources/projects/testGroovyProject/reference/pom.xml:26]: use property

**Comment by rbioteau** [comment]
⚠️ to be applied on https://github.com/bonitasoft/bonita-actorfilter-archetype

File: src/main/resources/archetype-resources/kotlin-pom.xml (hunk 2 above)
Comment (uguy) [reply src/main/resources/archetype-resources/kotlin-pom.xml]: fixed

**Pull request merged**